core/src/main/java/org/apache/carbondata/core/carbon/datastore/impl/btree/BTreeDataRefNodeFinder.java

```diff
@@ -240,9 +240,16 @@ private int compareIndexes(IndexKey first, IndexKey second) {
               firstNoDictionaryKeyBuffer.getShort(nonDictionaryKeyOffset + SHORT_SIZE_IN_BYTES);
           secondNodeDictionaryLength =
               secondNoDictionaryKeyBuffer.getShort(nonDictionaryKeyOffset + SHORT_SIZE_IN_BYTES);
-          compareResult = ByteUtil.UnsafeComparer.INSTANCE
-              .compareTo(first.getNoDictionaryKeys(), actualOffset, firstNoDcitionaryLength,
-                  second.getNoDictionaryKeys(), actualOffset, secondNodeDictionaryLength);
+          int minLength = Math.min(firstNoDcitionaryLength, secondNodeDictionaryLength);
```

Comment:
@gvramana Yes you right current logic has some problem we need to maintain offset of both the keys and when it is not the last key we need to substract the secondNodeDictionaryLength - actualOffset1 and firstNodeDictionaryLength-actualOffset.